### PR TITLE
fix(reader): stop reflowing message lines that already fit

### PR DESCRIPTION
**Pull request by robbiew**
Two FTN messages in the jam areas were rendering with their column alignment destroyed. Neither turned out to be an ANSI problem.

## Cause

`wrapAnsiString` (`internal/menu/executor_text_utils.go`) ran **every** body line through `strings.Fields(line)` and rejoined the words with a single space. That collapses runs of spaces and strips leading indent, so any message whose layout is built out of padding came apart. Quote, tear and origin lines were already exempted, which is why only the art rows looked wrong.

**Report 1** — a signature from Codefenix (21:4/141), Renegade pipe codes over CP437. Every row is 30 visible columns before the border, but the borders landed at columns 17, 29 and 28 instead of 31.

**Report 2** — the clrghouz new-node announcement (21:3/100), plain CP437 with no pipe codes and no escapes at all, every row padded to exactly 79 columns. The right-hand rail was crushed onto the text:

```
got  ": O : Hi Everyone, : O :"
want ": O : Hi Everyone,                                                        : O :"
```

## Fix

1. A line whose visible width already fits the terminal is emitted **verbatim**. No reflow, so signatures, tables and ASCII frames keep their spacing. This alone fixes both reports.
2. Lines that genuinely overflow go through a new `wrapVisualLine`, which splits into whitespace and word runs instead of tokenizing. Leading indent and interior spacing survive, only the whitespace a break actually lands on is consumed, and that whitespace's colour codes are carried onto the next line so a break cannot swallow an escape.

## Verification

Rendered both messages straight out of the live base (`/opt/v3/data/msgbases/fn.fsxnet_fsx_net`, msgs 326 and 329) through the real pipeline. The signature box now closes at column 31 on all five rows, and all 21 rows of the clrghouz frame come back byte-for-byte identical at both width 79 and 80.

The frame rows sit exactly at the width limit, which is the risky case for a "does it fit?" check. Two things make it safe: the reader passes the full terminal width (`message_reader.go:113`, default 80), not width−1; and the body is painted with absolute cursor positioning plus `\x1b[K` per line, so a full-width row cannot trigger the terminal's own auto-wrap. The news path writes CRLF-terminated lines instead, but it already budgets `termWidth-1` for that reason and is unaffected.

New `internal/menu/message_reader_wrap_test.go`, built from the real CP437 bytes — box alignment, interior spacing and indent, over-long lines still wrapping on word boundaries within budget with no words lost, colour surviving a break that swallows whitespace, and the full-width frame boundary case. All five were confirmed to fail against the pre-fix code and pass after. `go build ./...`, `go vet` and `go test ./...` are clean.

## Behaviour change worth knowing before deploy

Leading whitespace the old path silently dropped now reaches the screen, so some messages shift right a column or two — e.g. the Synchronet quote header in msg 329 now shows its authored two-space indent. I checked the JAM offsets to rule out an off-by-one on our side: msg 328 ends at 261639+452 = 262091, exactly where 329 begins, so those spaces are genuinely the sender's bytes.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved message and news wrapping to preserve indentation, spacing, alignment, and exact-width lines.
  * ANSI color formatting now remains consistent across wrapped lines.
  * Width calculations now account for the active terminal mode, including UTF-8 display widths and CP437 character widths.
  * Overlong words and lines continue to wrap without losing formatting.

* **Tests**
  * Added coverage for spacing, alignment, color preservation, character widths, terminal modes, and frame formatting.

### CODE REVIEW

File: internal/menu/executor_text_utils.go [hunk 1]
```diff
@@ -97,55 +97,156 @@ func wrapAnsiString(text string, width int) []string {
...
-					currentLine = ""
-					currentWidth = 0
-				} else {
-					currentLine = word
-					currentWidth = wordWidth
-				}
+		// A line that already fits is left exactly as the author typed it.
+		// Re-flowing it would collapse the runs of spaces that column-aligned
+		// signatures, tables and ASCII boxes are built out of.
+		if len(plainLine) <= width {
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/menu/executor_text_utils.go]: Fixed in 38f63d5, though not quite as suggested. `visibleWidth`/rune counting unconditionally would regress the case this PR exists to fix: CP437 art is one column per *byte*, and many adjacent CP437 pairs form a structurally valid UTF-8 sequence (writer.go documents this for #280 — `▄│` is DC B3, which decodes as U+0733). Counting runes there would undercount art rows and let over-wide lines through.

So the encoding is now decided once per line with `utf8.ValidString`, mirroring how `writeUTF8Mode` resolves the same ambiguity per span, and threaded into `splitWrapSegments` so segment widths advance by rune or by byte to match. Two tests pin both directions: a UTF-8 line that fits in columns but not bytes passes through verbatim, and a CP437 row stays measured in bytes.

File: internal/menu/executor_text_utils.go [hunk 2]
```diff
@@ -97,55 +97,156 @@ func wrapAnsiString(text string, width int) []string {
...
+		if curWidth > 0 && curWidth+seg.width > width {
+			flush()
+		}
+		cur.WriteString(seg.text)
+		curWidth += seg.width
+	}
+
+	if curWidth > 0 {
+		out = append(out, cur.String())
+	}
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/menu/executor_text_utils.go]: Confirmed and fixed in 38f63d5 — traced it exactly as described: a word of precisely `width` columns followed by ` \x1b[0m` flushes the visible text, leaves `cur` holding only the codes, and the `curWidth > 0` guard then drops them.

Took the suggested approach: the zero-width tail is appended to the last emitted line rather than becoming a row of its own, so the reset survives without shifting the body down a line. Covered by `TestMessageBodyKeepsTrailingResetAfterABreak`, which fails on the parent commit.

File: internal/menu/executor_text_utils.go [hunk 3]
```diff
@@ -97,55 +97,188 @@ func wrapAnsiString(text string, width int) []string {
...
+
+// columnWidth is the on-screen width of escape-free text. Message bodies are
+// raw bytes: CP437 art is one column per byte, while a UTF-8 body is one
+// column per rune. The two encodings cannot be told apart byte by byte -
+// plenty of adjacent CP437 pairs form a valid UTF-8 sequence - so the caller
+// decides once per line and passes the answer down, matching how
+// terminalio's writer resolves the same ambiguity for a span.
+func columnWidth(plain string, asUTF8 bool) int {
+	if asUTF8 {
+		return utf8.RuneCountInString(plain)
```
[the remark this answers is not in the document]
Comment (robbiew) [reply internal/menu/executor_text_utils.go]: Good catch — taken in d8ee1a8. `columnWidth` now uses `runewidth.StringWidth` for UTF-8 and `splitWrapSegments` advances by `runewidth.RuneWidth`, with CP437 still one column per byte. `TestMessageBodyMeasuresWideRunesAsTwoColumns` pins it: a line of three CJK words is 14 runes but 26 columns, so it fits by rune count and overruns a 20-column budget — it fails on the parent commit and passes now.

Two notes for the record. `go-runewidth v0.0.19` was already a direct dependency, so nothing new was pulled in. And this does leave `wrapAnsiString` measuring width differently from the rest of the codebase: `ansi.VisibleLength` and `news_linebreak.visibleWidth` both count one column per rune, and `executor_input_helpers.go:120` already documents the resulting CJK shortcoming. That inconsistency is pre-existing and harmless here — body lines are wrapped and written, never padded against those helpers — but making the codebase agree is worth its own pass rather than a drive-by in this PR.

[the remark this answers is not in the document]
Comment (robbiew) [reply internal/menu/executor_text_utils.go]: Thanks — no need, I have filed it: #363 covers the `ansi.VisibleLength` / `news_linebreak.visibleWidth` divergence, including the `breakOversizedLines` case inside the news path and the existing CJK comment at `executor_input_helpers.go:120`. #362 covers the separate ANSI-state-on-scroll issue Copilot raised.

Your outside-diff finding on `executor_text_utils.go:106-114` was correct and is fixed in 115a075 — `wrapAnsiString` now takes the output mode. One correction to the suggested remedy, though: "one column per byte for CP437 output" is not right for a valid UTF-8 span. `WriteStringCP437` folds each *rune* to a single CP437 byte (or `?` where it does not map), so in CP437 mode such a span is one column per **rune**, not per byte — byte counting would overcount a UTF-8 body viewed on a CP437 terminal, e.g. `café` is 5 bytes but renders as 4 columns.

So the rule implemented is: not valid UTF-8 → CP437, one column per byte in either mode; valid UTF-8 → display width on a UTF-8 terminal, one column per rune on a CP437 one. `TestMessageBodyMeasuresForTheOutputTerminal` pins your exact example — 79 chars plus `0xDC 0xB3` measures 80 columns in UTF-8 mode and 81 in CP437 mode, and wraps only in the latter. It fails on the parent commit.
